From 26c950e4937338d2515be70e8610905a0e022463 Mon Sep 17 00:00:00 2001
From: Shivam Mathur <shivam_jpr@hotmail.com>
Date: Mon, 25 Dec 2023 14:54:44 +0000
Subject: [PATCH] Fix ZEND_SIGNED_MULTIPLY_LONG for AArch64

Register operands are required in this context.

This is a backport of 8c8679207ae5ada8751288e75b78c928a4d34d1d
to PHP 5.6. The current code caused incorrect behavior on AArch64
when compiling with clang.
---
 Zend/zend_multiply.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Zend/zend_multiply.h b/Zend/zend_multiply.h
index a4d48b18918..850c1081f30 100644
--- a/Zend/zend_multiply.h
+++ b/Zend/zend_multiply.h
@@ -62,8 +62,8 @@
 	__asm__("mul %0, %2, %3\n"										\
 		"smulh %1, %2, %3\n"										\
 		"sub %1, %1, %0, asr #63\n"									\
-			: "=X"(__tmpvar), "=X"(usedval)							\
-			: "X"(a), "X"(b));										\
+			: "=&r"(__tmpvar), "=&r"(usedval)						\
+			: "r"(a), "r"(b));										\
 	if (usedval) (dval) = (double) (a) * (double) (b);				\
 	else (lval) = __tmpvar;											\
 } while (0)
-- 
2.43.0

